core/src/main/scala/com/microsoft/azure/synapse/ml/core/utils/JarLoadingUtils.scala

```diff
@@ -24,17 +28,8 @@ object JarLoadingUtils {
   }
 
   private[ml] val AllClasses = {
-    ClassPath.from(getClass.getClassLoader)
```

Comment:
I dont think you need this caching logic here because a val on an object is only computed once, you can even make it lazy if you don't want it always getting loaded